core/src/syn/parser/idiom.rs

```diff
@@ -252,10 +255,56 @@ impl Parser<'_> {
 				self.pop_peek();
 				Part::All
 			}
+			t!("{") => {
+				self.pop_peek();
+				self.parse_destructure_part()?
+			}
 			_ => Part::Field(self.next_token_value()?),
 		};
 		Ok(res)
 	}
+	/// Parse the part after the `.{` in an idiom
+	pub fn parse_destructure_part(&mut self) -> ParseResult<Part> {
+		let start = self.last_span();
+		let mut destructured: Vec<DestructurePart> = Vec::new();
+		loop {
+			let field: Ident = self.next_token_value()?;
+			let part = match self.peek_kind() {
+				t!(":") => {
+					self.pop_peek();
+					DestructurePart::Aliased(field, self.parse_local_idiom()?)
+				}
+				t!(".") => {
+					self.pop_peek();
+					let found = self.peek_kind();
+					match self.parse_dot_part()? {
+						Part::All => DestructurePart::All(field),
+						Part::Destructure(v) => DestructurePart::Destructure(field, v),
+						_ => {
+							return Err(ParseError::new(
+								ParseErrorKind::Unexpected {
+									found,
+									expected: "a star or a destructuring",
+								},
+								self.last_span(),
+							))
+						}
+					}
+				}
+				_ => DestructurePart::Field(field),
+			};
+
+			destructured.push(part);
+
+			if !self.eat(t!(",")) {
+				// We've reached the end of the destructure
+				self.expect_closing_delimiter(t!("}"), start)?;
+				break;
+			}
+		}
```

Comment:
It now doesn't support trailing commas which it probably should since most delimited sequences in surrealql support trailing separators. The pattern I often use for delimited sequences is first check if the closing delimiter is the next token, quit if so, otherwise parse a value then check for the comma, if there is none require the next token to be the delimiter, otherwise continue the loop.

Reply:
I guess we could just add the following after that last check right?
```rust
if self.eat(t!("}")) {
    break;
}
```

Reply:
Addressed in 73c499f17b3ec4a3059f278d73cb0d07d62425d9